src/mon/Monitor.cc

```diff
@@ -2924,11 +2924,14 @@ void Monitor::log_health(
   }
 }
 
-void Monitor::get_cluster_status(stringstream &ss, Formatter *f)
+void Monitor::get_cluster_status(stringstream &ss, Formatter *f,
+				 MonSession *session)
 {
   if (f)
     f->open_object_section("status");
 
+  const std::vector<string>& fs_names = session->get_allowed_fs_names();
```

Comment:
`session->get_allowed_fs_names()` produces an rvalue. You should do this:

`const auto fs_names = ...`

or

`const auto&& fs_names = ...`

Reply:
(and everywhere else you do this)